gittrigger.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"bytes"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+const gitTriggerString = "@rainforest"
+
+type gitTrigger struct {
+	Trigger    string
+	LastCommit string
+}
+
+func newGitTrigger() (gitTrigger, error) {
+	newGit := gitTrigger{Trigger: gitTriggerString}
+	err := newGit.getLatestCommit()
+	if err != nil {
+		return gitTrigger{}, err
+	}
+	return newGit, nil
+}
+
+func (g *gitTrigger) getLatestCommit() error {
+	cmd := exec.Command("git", "log", "-1", "--pretty=%B")
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err := cmd.Run()
+	if err != nil {
+		return err
+	}
+	g.LastCommit = strings.TrimSpace(out.String())
+	return nil
+}
+
+func (g gitTrigger) checkTrigger() bool {
+	shouldTrigger := strings.Contains(g.LastCommit, g.Trigger)
+	return shouldTrigger
+}
+
+func (g gitTrigger) getTags() []string {
+	tagRegex, _ := regexp.Compile("#([\\w_-]+)")
```

Comment:
Yeah, it's to escape `\` so it's `\w` in regex, I guess I could avoid this escaping using backtick, but oh, well. ;)